static/app/bootstrap/bootstrapRequests.tsx

```diff
@@ -0,0 +1,200 @@
+// XXX(epurkhiser): Ensure the LatestContextStore is initialized before we set
+// the active org. Otherwise we will trigger an action that does nothing
+import 'sentry/stores/latestContextStore';
+
+import {useEffect} from 'react';
+
+import {type ApiResult, Client} from 'sentry/api';
+import OrganizationStore from 'sentry/stores/organizationStore';
+import ProjectsStore from 'sentry/stores/projectsStore';
+import TeamStore from 'sentry/stores/teamStore';
+import type {Organization, Team} from 'sentry/types/organization';
+import type {Project} from 'sentry/types/project';
+import parseLinkHeader from 'sentry/utils/parseLinkHeader';
+import {queryOptions, useQuery} from 'sentry/utils/queryClient';
+
+// 30 second stale time
+// Stale time decides if the query should be refetched
+const BOOTSTRAP_QUERY_STALE_TIME = 30 * 1000;
+
+// 10 minute gc time
+// gc time is how long it will stay (even if stale) in the query cache
+const BOOTSTRAP_QUERY_GC_TIME = 10 * 60 * 1000;
+
+export function useBootstrapOrganizationQuery(orgSlug: string | null) {
+  const organizationQuery = useQuery(getBootstrapOrganizationQueryOptions(orgSlug));
+
+  useEffect(() => {
+    if (organizationQuery.data) {
+      OrganizationStore.onUpdate(organizationQuery.data);
+    }
+    if (organizationQuery.error) {
+      OrganizationStore.onFetchOrgError(organizationQuery.error as any);
+    }
+  }, [organizationQuery.data, organizationQuery.error]);
+
+  return organizationQuery;
+}
+
+export function useBootstrapTeamsQuery(orgSlug: string | null) {
+  const teamsQuery = useQuery(getBoostrapTeamsQueryOptions(orgSlug));
+
+  useEffect(() => {
```

Comment:
```suggestion
  useLayoutEffect(() => {
```
Might save us some awkward flicker